### PR TITLE
[new package] zydis 4.1.0

### LINKED ISSUE #21138: Add mingw-w64-zydis
### Package name

mingw-w64-zydis

### Brief description of package

Fast and lightweight x86/x86-64 disassembler and code generation library

### URL for package's homepage

https://zydis.re/

### Provide a basic test case to validate the package's functionality.

 ./ZydisInfo.exe,

This builds very easily using cmake, this would be helpful for other tools like x64dbg

### MINGW environments where you need the package

- [X] MINGW64
- [ ] MINGW32
- [X] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

_No response_

### PR DESCRIPTION
**Pull request by ognevny**
fixes #21138

### CODE REVIEW

**Comment by ognevny** [comment]
what is that?
```
  C:/_/B/src/zydis/resources/VersionInfo.rc:43:75: warning: null character(s) ignored
     43 | <U+0000> <U+0000>*<U+0000> <U+0000>O<U+0000>U<U+0000>T<U+0000> <U+0000>O<U+0000>F<U+0000> <U+0000>O<U+0000>R<U+0000> <U+0000>I<U+0000>N<U+0000> <U+0000>C<U+0000>O<U+0000>N<U+0000>N<U+0000>E<U+0000>C<U+0000>T<U+0000>I<U+0000>O<U+0000>N<U+0000> <U+0000>W<U+0000>I<U+0000>T<U+0000>H<U+0000> <U+0000>T<U+0000>H<U+0000>E<U+0000> <U+0000>S<U+0000>O<U+0000>F<U+0000>T<U+0000>W<U+0000>A<U+0000>R<U+0000>E<U+0000> <U+0000>O<U+0000>R<U+0000> <U+0000>T<U+0000>H<U+0000>E<U+0000> <U+0000>U<U+0000>S<U+0000>E<U+0000> <U+0000>O<U+0000>R<U+0000> <U+0000>O<U+0000>T<U+0000>H<U+0000>E<U+0000>R<U+0000> <U+0000>D<U+0000>E<U+0000>A<U+0000>L<U+0000>I<U+0000>N<U+0000>G<U+0000>S<U+0000> <U+0000>I<U+0000>N<U+0000> <U+0000>T<U+0000>H<U+0000>E<U+0000>
        |                                                                                                                                                                                                                                                                                                                                              ^~~~~~~~
```

**Comment by ognevny** [comment]
looks like rc file is broken somehow

**Comment by mmuetzel** [comment]
Maybe that is a UTF-16 encoded file being interpreted as UTF-8 (or CP1252 or a similar encoding).

**Comment by ognevny** [comment]
> Maybe that is a UTF-16 encoded file being interpreted as UTF-8 (or CP1252 or a similar encoding).

yes, they used UTF-16 encoding for a file. I downloaded a new version with UTF-8 encoding (it was fixed 2 months ago, the latest release was 5 months ago)

File: mingw-w64-zydis/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+             'git')
+source=("${url}/archive/v${pkgver}/${_realname}-${pkgver}.tar.gz"
+        "https://raw.githubusercontent.com/netadr/zydis/44eefca6ad7b124f0fce234483dcb555ce7b2a0e/resources/VersionInfo.rc")
+sha256sums=('31f23de8abb4cc2efa0fd0e827bbabcaa0f3d00fcaed8598e05295ba7b3806ad'
+            '4e4822ed30a7ec62e87d9628cc8215c7000f898e86f8e51cce6eb57fc5159404')
+
+prepare() {
+  cd "${_realname}-${pkgver}"
+
+  cp -f "${srcdir}/VersionInfo.rc" resources/VersionInfo.rc
```
Comment (Biswa96) [inline mingw-w64-zydis/PKGBUILD:28]:
```suggestion
  dos2unix "${_realname}/resources/VersionInfo.rc"
```

No need to copy it from another commit.

**Pull request merged**